packages/ui/src/components/pagination/pagination-primitive.tsx

```diff
@@ -37,27 +37,34 @@ type PaginationPrimitiveLinkGeneralProps = {
 
 interface PaginationPrimitiveLinkProps extends Omit<PaginationPrimitiveLinkGeneralProps, 't'> {
   isFullRounded?: boolean
+  disabled?: boolean
+  variant?: ButtonVariants
+  theme?: ButtonThemes
 }
 
 const PaginationPrimitiveLink = ({
   className,
   isActive,
-  size,
-  isFullRounded = true,
+  size = 'sm',
+  isFullRounded: rounded,
   children,
+  disabled = false,
+  variant = 'soft',
+  theme = 'muted',
   ...props
 }: PaginationPrimitiveLinkProps) => (
   <a
     aria-current={isActive ? 'page' : undefined}
     className={cn(
       buttonVariants({
-        variant: 'custom',
-        size: size ? size : 'sm_icon',
-        borderRadius: isFullRounded ? 'full' : 'default'
+        variant,
+        size,
+        rounded,
+        theme
       }),
-      'text-1 bg-cn-background-2 text-cn-foreground-1 hover:bg-cn-background-3 w-auto min-w-7 px-1.5 font-normal',
       {
-        'bg-cn-background-8 shadow-1 hover:bg-cn-background-8 cursor-default': isActive
+        'button-active': isActive,
```

Comment:
`button-active` and `button-disabled` classes have been added to button only for the pagination purpose. Once pagination component is ready, these classes will be removed.